ppb_vector/vector2.py

```diff
@@ -241,10 +242,21 @@ def isclose(self: VectorOrSub, other: VectorLike, *,
             diff < float(abs_tol)
         )
 
-    def rotate(self: VectorOrSub, degrees: Realish) -> VectorOrSub:
-        r = radians(degrees)
-        r_cos = cos(r)
-        r_sin = sin(r)
+    @staticmethod
+    def trig(angle: Realish):
+        from math import copysign, cos, sin, radians, sqrt
+        r = radians(angle)
+        r_cos, r_sin = cos(r), sin(r)
+
+        if abs(r_cos) > abs(r_sin):
+            r_sin = copysign(sqrt(1 - r_cos * r_cos), r_sin)
+        else:
+            r_cos = copysign(sqrt(1 - r_sin * r_sin), r_cos)
```

Comment:
Ok, what is going on here?

Reply:
I pushed a commit that add comments.  Basically, it keeps the largest value (our of `cos(r)` and `sin(r)`) and recomputes the other one from the equation `cos²(r) + sin²(r) = 1`.
Since the value of `r_cos² + r_sin²` determines how well we are preserving vector lengths, it means vector lengths are more stable under rotations, which solves the issue in #87